extensions/oplogPopulator/modules/Connector.js

```diff
@@ -260,7 +286,16 @@ class Connector {
         this._buckets.delete(bucket);
         this._updateConnectorState(true);
         try {
-            await this.updatePipeline(doUpdate);
+            if (this._isPipelineImmutable && this._buckets.size >= 1) {
+                this._logger.warn('Removing a bucket from an immutable pipeline', {
+                    method: 'Connector.removeBucket',
+                    connector: this._name,
```

Comment:
All oplog populator metrics but two have a `connector` label... I created [BB-604](https://scality.atlassian.net/browse/BB-604).

Reply:
> I think we already have one case where we detect that we do not "respect" the config (e.g. when the number of connectors is reduced)

We have the `s3_oplog_populator_connectors` metric that we decrement when removing a connector, so we can see that we do not respect the config indeed. If we want to have metrics do detect that we have a deviation, we will need to have a metric per connector, otherwise, it'll just be a counter of the number of "not modified" connectors (something like `s3_oplog_populator_connector_reconfiguration_skipped`). Do we have any preference?

Reply:
we could have a _gauge_ on the number of buckets "unexpectedly listened to", i.e. present in multi-bucket connector config from earlier version but not present anymore.